corehq/apps/export/views/list.py

```diff
@@ -536,13 +541,20 @@ def page_context(self):
         }
 
 
-def _get_task_status_json(export_instance_id):
-    status = get_saved_export_task_status(export_instance_id)
+def _get_task_status_json(export_instance):
+    status = get_saved_export_task_status(export_instance._id)
+    failure_reason = None
+    if status.failed():
```

Comment:
It looks like `status` has an `error` and a `result` field. Is it possible to get this info from there so we don't have to check the size each time? That way if the size changes between the attempt and when they view the page, the user will also get accurate info.

Reply:
Great point. After looking into the `error` field, it only saves the string output of the Exception raised. I could have keyed off of a custom exception name in this string, but that felt brittle, so I instead added an `exception` field to `TaskStatus` that is populated when applicable, otherwise is None.

This way, we can compare `task.exception` to an actual reference to the custom exception class which is far less brittle.

See 4452d953e11e8d393e963eed85d938d0ad6b599d